helm/hpcc/templates/_warnings.tpl

```diff
@@ -126,22 +75,58 @@ Pass in dict with root and warnings
    {{- $_ := set $warning "msg" (printf "Default cost parameters are being used for the storage %s: %s" ((len $ctx.planesWithDefaultCosts)|plural "plane" "planes") ($ctx.planesWithDefaultCosts|toStrings)) -}}
    {{- $_ := set $ctx "warnings" (append $ctx.warnings $warning) -}}
  {{- end -}}
- {{- /* Ensure that resources have been provided for all components */ -}}
- {{- /* Build on the existing items in $ctx.missingResources created previously for thor, eclccserver and eclagent */ -}}
- {{- $components := omit (ternary (omit .root.Values "sasha") .root.values .root.Values.sasha.disabled) "global" "security" "storage" "certificates" "secrets" "vaults" "bundles" "thor" "eclccserver" "eclagent" -}}
- {{- range $ctypes := $components -}}
+ {{- /* Warn when resources not provided, default cpu rate used and components requiring resources for cost calcs */ -}}
+ {{- if eq .root.Values.global.cost.perCpu 0.0565000000001 -}}
+  {{- $_ := set $ctx "usingDefaultCpuCost" "true" -}}
+ {{- end -}}
+ {{- $_ := set $ctx "missingResources" list -}}
+ {{- $_ := set $ctx "missingResourcesForCosts" list -}}
+ {{- $_ := set $ctx "defaultCpuRateComponents" list -}}
+ {{-  $_ := set $ctx "components" (pick .root.Values "dafilesrv" "dali" "sasha" "dfuserver" "eclagent" "eclccserver" "esp" "roxie" "thor" "eclscheduler") -}}
+ {{- if (.root.Values.sasha.disabled|default false) -}}
+  {{- $_ := set $ctx "components" (omit $ctx.components "sasha") -}}
+ {{- end -}}
+ {{- range $cname, $ctypes := $ctx.components -}}
   {{- range $id, $component := $ctypes -}}
    {{- if not $component.disabled -}}
-    {{- $hasResources := include "hpcc.hasResources" (dict "resources" $component.resources) -}}
+    {{- $hasResources := "" -}}
+    {{- if eq $cname "thor" -}}
+     {{- $hasResources = include "hpcc.hasResources" (dict "resources" $component.managerResources) -}}
```

Comment:
@jakesmith You mentioned before that = shouldn't be used because it is unsupported feature.  It was actually introduced into the language in a March 2022 release.  It's documented here: https://pkg.go.dev/text/template@go1.18.  Helm uses go1.18 as of version v3.10.0.

Reply:
@jakesmith should we switch to using it?  It would simplify some of the code.

Reply:
NB: we did try to detect the helm version and check minimum version before: https://track.hpccsystems.com/browse/HPCC-27088